src/Kiota.Builder/CodeRenderer.cs

```diff
@@ -21,31 +26,36 @@ public static async Task RenderCodeNamespaceToSingleFileAsync(LanguageWriter wri
-        public static async Task RenderCodeNamespaceToFilePerClassAsync(LanguageWriter writer, CodeNamespace root, bool shouldWriteNamespaceIndices, string namespacePrefix)
+        public async Task RenderCodeNamespaceToFilePerClassAsync(LanguageWriter writer, CodeNamespace root)
         {
             foreach (var codeElement in root.GetChildElements(true))
             {
                 if (codeElement is CodeClass codeClass)
                     await RenderCodeNamespaceToSingleFileAsync(writer, codeClass, writer.PathSegmenter.GetPath(root, codeClass));
                 else if (codeElement is CodeEnum codeEnum)
                     await RenderCodeNamespaceToSingleFileAsync(writer, codeEnum, writer.PathSegmenter.GetPath(root, codeEnum));
-                else if(codeElement is CodeNamespace codeNamespace) {
-                    
-                    if(!string.IsNullOrEmpty(codeNamespace.Name) && !string.IsNullOrEmpty(root.Name) && shouldWriteNamespaceIndices && !namespacePrefix.Contains(codeNamespace.Name, StringComparison.OrdinalIgnoreCase)) {
+                else if (codeElement is CodeNamespace codeNamespace)
+                {
+                    if (!string.IsNullOrEmpty(codeNamespace.Name) && !string.IsNullOrEmpty(root.Name) &&
+                        _configuration.ShouldWriteNamespaceIndices &&
+                        !_configuration.ClientNamespaceName.Contains(codeNamespace.Name, StringComparison.OrdinalIgnoreCase))
+                    {
                         var namespaceNameLastSegment = codeNamespace.Name.Split('.').Last().ToLowerInvariant();
-                        // for ruby if the module already has a class with the same name, it's going to be declared automatically
-                        if(codeNamespace.FindChildByName<CodeClass>(namespaceNameLastSegment, false) == null)
+                        // if the module already has a class with the same name, it's going to be declared automatically
+                        if (_configuration.ShouldWriteBarrelsIfClassExists ||
+                            codeNamespace.FindChildByName<CodeClass>(namespaceNameLastSegment, false) == null)
                             await RenderCodeNamespaceToSingleFileAsync(writer, codeNamespace, writer.PathSegmenter.GetPath(root, codeNamespace));
                     }
-                    await RenderCodeNamespaceToFilePerClassAsync(writer, codeNamespace, shouldWriteNamespaceIndices, namespacePrefix);
+                    await RenderCodeNamespaceToFilePerClassAsync(writer, codeNamespace);
                 }
             }
         }
-        private static readonly CodeElementOrderComparer rendererElementComparer = new CodeElementOrderComparer();
-        private static void RenderCode(LanguageWriter writer, CodeElement element)
+        private readonly CodeElementOrderComparer rendererElementComparer;
```

Comment:
nit: incosistent naming of private readonly variables. Should probably be `_rendererElementComparer`

Reply:
fixed